src/core/resource.h

```diff
@@ -148,6 +148,15 @@ class CResource : public CFunctionBlock, public forte::core::CFBContainer{
       return nullptr;
     }
 
+    struct InitValue {
```

Comment:
i would recommend, making this a regular class, with getters, setters, a ctor setting the two fields and making the fields const (assuming they are never changed after set at the beginning).
Then you can add a move operator for efficient insertion into the saved InitalValues.
Also, please rename the fields. A string called "str" repeats only its type information, but not its intend. I assume its a name of some sort, so please change the variable name to something suitable. Same for the pointer called ptr. How about iecVariable or similar here?


Personally I am not a big fan of nested classes, as the"advantage" of reduced visibility is usually not worth the cost of later maintenance (especially as visibility is also controlled by includes). If for some reason, this is later needed in other context, the additional work of extracting this can be substantial

Reply:
I believe (the renamed) `ptr` could then also be a reference, which would save on `nullptr` checks.